### ISSUE TITLE
contract was not deployed

### ISSUE THREAD

**Issue by fedealconada** [matter-labs/foundry-zksync#685]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.0.2 (ac5feb1 2024-10-30T00:25:47.390509000Z)

### What command(s) is the bug in?

forge create

### Operating System

macOS (Apple Silicon)

### Describe the bug

Trying to deploy a contract using `forge create` and `zk-paymaster-address` returns `contract was not deployed`:

```
forge create ./contracts/v2/FiatTokenV2_2.sol:FiatTokenV2_2 --rpc-url testnet --private-key $DEPLOYER_PRIVATE_KEY --zksync --libraries ./contracts/util/SignatureChecker.sol:SignatureChecker:0x1D51129dA614f83bE04511145B24bA90F5A85491 --zk-paymaster-address PAYMASTER_ADDRESS --zk-paymaster-input $(cast calldata "general(bytes)" "0x")
```

**Comment by Jrigada** [comment]
Hello @fedealconada. I cloned [this](https://github.com/circlefin/stablecoin-evm) repo and I wasn't able to compile the given project, so I had to use a newer, yet to be released compiler version 1.5.7 and I was able to successfully deploy with your command using Sophon's paymaster address.
[Here](https://explorer.testnet.sophon.xyz/address/0x3C4E4A6ef46706F387afEe662Fc53700B24c8230#contract) is the deployed contract and [here](https://explorer.testnet.sophon.xyz/tx/0xbc9ee907706aafb82dd149b40e1b4d3eaabf6935a1aea50117c63d6b968c8777) is the tx detailing the use of the paymaster. 
![Image](https://github.com/user-attachments/assets/8dd94321-721a-4a42-96d1-89e22926710e)

Could you try running the deployment again to rule out an RPC issue or any other intermittent error?

P.S.: We are about to merge the PR that adds support for zksolc 1.5.7 to make the process easier.

**Comment by dutterbutter** [comment]
Closing due to staleness
